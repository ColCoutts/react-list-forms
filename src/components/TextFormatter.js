import React, { PureComponent } from 'react';

export default class TextFormatter extends PureComponent {
  state = {
    text: '',
    color: 'black'
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name] : target.value });
  }

  render() {
    const { text, color } = this.state;

    return (
      <>
        <input name="text" value={text} onChange={this.handleChange}></input>
        <input name="color" type="color" value={color} onChange={this.handleChange}></input>
        <h1 style={{ color }}>{text}</h1>
      </>
    );
  }
}
