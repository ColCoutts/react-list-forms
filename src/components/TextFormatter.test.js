import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('textFormatter component', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
