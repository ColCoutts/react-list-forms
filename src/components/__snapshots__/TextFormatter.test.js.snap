// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textFormatter component renders a snapshot 1`] = `
<Fragment>
  <input
    name="text"
    onChange={[Function]}
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="black"
  />
  <h1
    style={
      Object {
        "color": "black",
      }
    }
  />
</Fragment>
`;
